In [2]:
from numpy import sqrt
import pandas as pd
from IPython.display import Markdown, display
import scipy.stats as st

df = pd.read_csv('anggur.csv')
df.head()

fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0           5.90            0.4451       0.1813        2.049401   0.070574   
1           8.40            0.5768       0.2099        3.109590   0.101681   
2           7.54            0.5918       0.3248        3.673744   0.072416   
3           5.39            0.4201       0.3131        3.371815   0.072755   
4           6.51            0.5675       0.1940        4.404723   0.066379   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0            16.593818                 42.27   0.9982  3.27       0.71   
1            22.555519                 16.01   0.9960  3.35       0.57   
2             9.316866                 35.52   0.9990  3.31       0.64   
3            18.212300                 41.97   0.9945  3.34       0.55   
4             9.360591                 46.27   0.9925  3.27       0.45   

   alcohol  quality  
0     8.64        7  
1    10.03        8  
2     9.23        8  
3    14.07        9  
4    11.49        8

## 1. Menulis deskripsi statistika (Descriptive Statistics) dari semua kolom pada data yang bersifat numerik, terdiri dari mean, median, modus, standar deviasi, variansi, range, nilai minimum, maksimum, kuartil, IQR, skewness dan kurtosis.


In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 12 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   fixed acidity         1000 non-null   float64
 1   volatile acidity      1000 non-null   float64
 2   citric acid           1000 non-null   float64
 3   residual sugar        1000 non-null   float64
 4   chlorides             1000 non-null   float64
 5   free sulfur dioxide   1000 non-null   float64
 6   total sulfur dioxide  1000 non-null   float64
 7   density               1000 non-null   float64
 8   pH                    1000 non-null   float64
 9   sulphates             1000 non-null   float64
 10  alcohol               1000 non-null   float64
 11  quality               1000 non-null   int64  
dtypes: float64(11), int64(1)
memory usage: 93.9 KB


#### Semua kolom data anggur sudah bersifat numerik

In [37]:
desc_stats_df = pd.DataFrame(columns=df.columns)
desc_stats_df.loc['Mean'] = df.mean()
desc_stats_df.loc['Median'] = df.median()
desc_stats_df.loc['Modus'] = df.mode().loc[0]
desc_stats_df.loc['Std. Deviation'] = df.std()
desc_stats_df.loc['Variance'] = df.var()
desc_stats_df.loc['Minimum'] = df.min()
desc_stats_df.loc['Q1'] = df.quantile(0.25)
desc_stats_df.loc['Q2'] = df.quantile(0.5)
desc_stats_df.loc['Q3'] = df.quantile(0.75)
desc_stats_df.loc['Maximum'] = df.max()
desc_stats_df.loc['Range'] = desc_stats_df.loc['Maximum'] - desc_stats_df.loc['Minimum']
desc_stats_df.loc['IQR'] = desc_stats_df.loc['Q3'] - desc_stats_df.loc['Q1']
desc_stats_df.loc['Skewness'] = df.skew()
desc_stats_df.loc['Kurtosis'] = df.kurtosis()
desc_stats_df

fixed acidity  volatile acidity  citric acid  residual sugar  \
Mean                 7.152530          0.520839     0.270517        2.567104   
Median               7.150000          0.524850     0.272200        2.519430   
Modus                6.540000          0.554600     0.301900        0.032555   
Std. Deviation       1.201598          0.095848     0.049098        0.987915   
Variance             1.443837          0.009187     0.002411        0.975977   
Minimum              3.320000          0.139900     0.116700        0.032555   
Q1                   6.377500          0.456100     0.237800        1.896330   
Q2                   7.150000          0.524850     0.272200        2.519430   
Q3                   8.000000          0.585375     0.302325        3.220873   
Maximum             11.490000          0.805100     0.409600        5.550755   
Range                8.170000          0.665200     0.292900        5.518200   
IQR                  1.622500          0.129275     0.064525        1.324544   
Skewness            -0.028879         -0.197699    -0.045576        0.132638   
Kurtosis            -0.019292          0.161853    -0.104679       -0.042980   

                chlorides  free sulfur dioxide  total sulfur dioxide  \
Mean             0.081195            14.907679             40.290150   
Median           0.082167            14.860346             40.190000   
Modus            0.015122             0.194679             35.200000   
Std. Deviation   0.020111             4.888100              9.965767   
Variance         0.000404            23.893519             99.316519   
Minimum          0.015122             0.194679              3.150000   
Q1               0.066574            11.426717             33.785000   
Q2               0.082167            14.860346             40.190000   
Q3               0.095312            18.313098             47.022500   
Maximum          0.140758            27.462525             69.960000   
Range            0.125635            27.267847             66.810000   
IQR              0.028738             6.886381             13.237500   
Skewness        -0.051319             0.007130             -0.024060   
Kurtosis        -0.246508            -0.364964              0.063950   

                 density        pH  sulphates    alcohol    quality  
Mean            0.995925  3.303610   0.598390  10.592280   7.958000  
Median          0.996000  3.300000   0.595000  10.610000   8.000000  
Modus           0.995900  3.340000   0.590000   9.860000   8.000000  
Std. Deviation  0.002020  0.104875   0.100819   1.510706   0.902802  
Variance        0.000004  0.010999   0.010164   2.282233   0.815051  
Minimum         0.988800  2.970000   0.290000   6.030000   5.000000  
Q1              0.994600  3.230000   0.530000   9.560000   7.000000  
Q2              0.996000  3.300000   0.595000  10.610000   8.000000  
Q3              0.997200  3.370000   0.670000  11.622500   9.000000  
Maximum         1.002600  3.710000   0.960000  15.020000  10.000000  
Range           0.013800  0.740000   0.670000   8.990000   5.000000  
IQR             0.002600  0.140000   0.140000   2.062500   2.000000  
Skewness       -0.076883  0.147673   0.149199  -0.018991  -0.089054  
Kurtosis        0.016366  0.080910   0.064819  -0.131732   0.108291

In [47]:
display(Markdown("##### Terdapat beberapa kolom dengan nilai modus yang lebih dari satu:"))

multi_modus = df.mode()
multi_modus = multi_modus.loc[:, pd.notna(multi_modus.loc[1])]
display(multi_modus)

##### Terdapat beberapa kolom dengan nilai modus yang lebih dari satu:

residual sugar  chlorides  free sulfur dioxide  total sulfur dioxide  \
0          0.032555   0.015122             0.194679                 35.20   
1          0.033333   0.020794             0.621628                 37.25   
2          0.051774   0.024259             0.860177                 39.64   
3          0.077156   0.027209             3.032139                 40.61   
4          0.084744   0.032111             3.129885                 41.05   
..              ...        ...                  ...                   ...   
995        5.210260   0.131425            26.630490                   NaN   
996        5.217429   0.133656            26.665773                   NaN   
997        5.252864   0.135368            26.822626                   NaN   
998        5.299524   0.135790            27.006307                   NaN   
999        5.550755   0.140758            27.462525                   NaN   

     density  alcohol  
0     0.9959     9.86  
1     0.9961    10.31  
2     0.9965      NaN  
3     0.9970      NaN  
4        NaN      NaN  
..       ...      ...  
995      NaN      NaN  
996      NaN      NaN  
997      NaN      NaN  
998      NaN      NaN  
999      NaN      NaN  

[1000 rows x 6 columns]

## 4. Melakukan test hipotesis 1 sampel

### a. Nilai rata-rata pH di atas 3.29?

1. $H_0$: $μ = 3.29$
2. $H_1$: $μ > 3.29$
3. $α = 0.05$
4. Uji statistik: 
$$z = \frac{\bar{x} - μ_0}{σ/\sqrt{n}} $$
&emsp;&emsp; Dengan critical region:
$$ z > z_{\alpha} $$
$$ z > 1.645 $$
5. Dapat dihitung:

In [4]:
x_bar = df['pH'].mean()
mu = 3.29
sigma = df['pH'].std()
n = df['pH'].count()

z_value = (x_bar - mu) * (n**(0.5)) / (sigma)
p_value = 1 - st.norm.cdf(z_value)

display(Markdown(f"Diketahui: <br> $\\bar{{x}} = {x_bar}$ &emsp; $\\mu = {mu}$ <br> $\\sigma = {sigma:.5f}$ &emsp; $n = {n}$"))
display(Markdown(f"Didapatkan nilai: <br>$z = {z_value:.3f} $ <br> $p = {p_value} $"))

Diketahui: <br> $\bar{x} = 3.30361$ &emsp; $\mu = 3.29$ <br> $\sigma = 0.10488$ &emsp; $n = 1000$

Didapatkan nilai: <br>$z = 4.104 $ <br> $p = 2.032263004325685e-05 $

6. Karena $z = 4.104 > z_{\alpha/2}$, $z$ berada di critical section sehingga <u>hipotesis $H_0$ ditolak</u>. 
<br> Dengan tes signifikan, disimpulkan bahwa <u>hipotesis $H_0$ ditolak</u> karena <i>p-value</i> lebih kecil dari nilai signifikan yang ditetapkan ($p < 0.05$).
<br> Dapat disimpulkan bahwa rata-rata pH lebih besar dari 3.29.

### b. Nilai rata-rata Residual Sugar tidak sama dengan 2.50?

1. $H_0$: $μ = 2.50$
2. $H_1$: $μ \neq 2.50$ &emsp; ($\mu < 2.50$ atau    $\mu > 2.50$)
3. $α = 0.05$
4. Uji statistik: 
$$z = \frac{\bar{x} - μ_0}{σ/\sqrt{n}} $$
&emsp;&emsp; Dengan critical region:
$$ z < -z_{\alpha/2} \quad atau \quad z > z_{\alpha/2} $$
$$ z < -1.96 \quad atau \quad z > 1.96 $$
5. Dapat dihitung:

In [5]:
x_bar = df['residual sugar'].mean()
mu = 2.5
sigma = df['residual sugar'].std()
n = df['residual sugar'].count()

z_value = (x_bar - mu) * (n**(0.5)) / (sigma)
p_value = 2*(1 - st.norm.cdf(z_value))

display(Markdown(f"Diketahui: <br> $\\bar{{x}} = {x_bar}$ &emsp; $\\mu = {mu}$ <br> $\\sigma = {sigma:.5f}$ &emsp; $n = {n}$"))
display(Markdown(f"Didapatkan nilai: <br>$z = {z_value:.3f} $ <br> $p = {p_value} $"))

Diketahui: <br> $\bar{x} = 2.5671036825069997$ &emsp; $\mu = 2.5$ <br> $\sigma = 0.98792$ &emsp; $n = 1000$

Didapatkan nilai: <br>$z = 2.148 $ <br> $p = 0.03171677881789692 $

6. Karena $z = 2.148 > z_{\alpha/2}$, $z$ berada di critical section sehingga <u>hipotesis $H_0$ ditolak</u>. 
<br> Dengan tes signifikan, disimpulkan bahwa <u>hipotesis $H_0$ ditolak</u> karena <i>p-value</i> lebih kecil dari nilai signifikan yang ditetapkan ($p < 0.05$).
<br> Dapat disimpulkan bahwa nilai residual sugar tidak sama dengan 2.148.

### c. Nilai rata-rata 150 baris pertama kolom sulphates bukan 0.65?

1. $H_0$: $μ = 0.65$
2. $H_1$: $μ \neq 0.65$ &emsp; ($\mu < 0.65$ atau    $\mu > 0.65$)
3. $α = 0.05$
4. Uji statistik: 
$$z = \frac{\bar{x} - μ_0}{σ/\sqrt{n}} $$
&emsp;&emsp; Dengan critical region:
$$ z < -z_{\alpha/2} \quad atau \quad z > z_{\alpha/2} $$
$$ z < -1.96 \quad atau \quad z > 1.96 $$
5. Dapat dihitung:

In [6]:
x_bar = df.loc[:150, 'sulphates'].mean()
mu = 0.65
sigma = df.loc[:150, 'sulphates'].std()
n = df.loc[:150, 'sulphates'].count()

z_value = (x_bar - mu) * (n**(0.5)) / (sigma)
p_value = 2*(1 - st.norm.cdf(abs(z_value)))

display(Markdown(f"Diketahui: <br> $\\bar{{x}} = {x_bar}$ &emsp; $\\mu = {mu}$ <br> $\\sigma = {sigma:.5f}$ &emsp; $n = {n}$"))
display(Markdown(f"Didapatkan nilai: <br>$z = {z_value:.3f} $ <br> $p = {p_value} $"))

Diketahui: <br> $\bar{x} = 0.6064900662251657$ &emsp; $\mu = 0.65$ <br> $\sigma = 0.10878$ &emsp; $n = 151$

Didapatkan nilai: <br>$z = -4.915 $ <br> $p = 8.868285241714347e-07 $

6. Karena $z = -4.915 < z_{\alpha/2}$, $z$ berada di critical section sehingga <u>hipotesis $H_0$ ditolak</u>. 
<br> Dengan tes signifikan, disimpulkan bahwa <u>hipotesis $H_0$ ditolak</u> karena <i>p-value</i> lebih kecil dari nilai signifikan yang ditetapkan ($p < 0.05$).
<br> Dapat disimpulkan bahwa nilai rata-rata 150 baris pertama kolom sulphates bukan 0.65.

### d. Nilai rata-rata total sulfur dioxide di bawah 35?

1. $H_0$: $μ = 35$
2. $H_1$: $\mu < 35$
3. $α = 0.05$
4. Uji statistik: 
$$z = \frac{\bar{x} - μ_0}{σ/\sqrt{n}} $$
&emsp;&emsp; Dengan critical region:
$$ z < -z_{\alpha}$$
$$ z < -1.645$$
5. Dapat dihitung:

In [7]:
x_bar = df['total sulfur dioxide'].mean()
mu = 35
sigma = df['total sulfur dioxide'].std()
n = df['total sulfur dioxide'].count()

z_value = (x_bar - mu) * (n**(0.5)) / (sigma)
p_value = st.norm.cdf(z_value)

display(Markdown(f"Diketahui: <br> $\\bar{{x}} = {x_bar}$ &emsp; $\\mu = {mu}$ <br> $\\sigma = {sigma:.5f}$ &emsp; $n = {n}$"))
display(Markdown(f"Didapatkan nilai: <br>$z = {z_value:.3f} $ <br> $p = {p_value} $"))

Diketahui: <br> $\bar{x} = 40.290150000000004$ &emsp; $\mu = 35$ <br> $\sigma = 9.96577$ &emsp; $n = 1000$

Didapatkan nilai: <br>$z = 16.786 $ <br> $p = 1.0 $

6. Karena $z = 16.915 > z_{\alpha}$, $z$ tidak berada di critical section sehingga <u>hipotesis $H_0$ tidak ditolak</u>. 
<br> Dengan tes signifikan, disimpulkan juga bahwa <u>hipotesis $H_0$ tidak ditolak</u> karena <i>p-value</i> lebih besar dari nilai signifikan yang ditetapkan ($p > 0.05$).
<br> Tidak dapat disimpulkan bahwa nilai rata-rata total sulfur dioxide di bawah 35.

### e. Proporsi nilai total Sulfat Dioxide yang lebih dari 40, adalah tidak sama dengan 50%?

1. $H_0$: $p = 0.5$
2. $H_1$: $p \neq 0.5$ &emsp; ($p < 0.5$ atau    $p > 0.5$)
3. $α = 0.05$
4. Uji statistik: 
$$z = \frac{\hat{p} - p_0}{\sqrt{p_0q_0/n}} $$
&emsp;&emsp; Dengan critical region:
$$ z < -z_{\alpha/2} \quad atau \quad z > z_{\alpha/2} $$
$$ z < -1.96 \quad atau \quad z > 1.96 $$
5. Dapat dihitung:

In [8]:
p_0 = 0.5
q_0 = 1 - p_0
n = df['total sulfur dioxide'].count()
p_hat = df.loc[df['total sulfur dioxide'] > 40]['total sulfur dioxide'].count() / n

z_value = (p_hat - p_0) * (n ** 0.5) / ((p_0 * q_0) ** 0.5)
p_value = 2 * (1 - st.norm.cdf(abs(z_value)))

display(Markdown(f"Diketahui: <br> $p_0 = {p_0}$ $&emsp; $q_0 = {q_0}$ <br> $n = {n}$ &emsp; $\\hat{{p}} = {p_hat}$"))
display(Markdown(f"Didapatkan nilai: <br>$z = {z_value:.3f} $ <br> $p = {p_value} $"))

Diketahui: <br> $p_0 = 0.5$ $&emsp; $q_0 = 0.5$ <br> $n = 1000$ &emsp; $\hat{p} = 0.512$

Didapatkan nilai: <br>$z = 0.759 $ <br> $p = 0.4478844782641116 $

6. Karena $-z_{\alpha/2} < z = 0.759 < z_{\alpha/2}$, $z$ tidak berada di critical section sehingga <u>hipotesis $H_0$ tidak ditolak</u>. 
<br> Dengan tes signifikan, disimpulkan juga bahwa <u>hipotesis $H_0$ tidak ditolak</u> karena <i>p-value</i> lebih besar dari nilai signifikan yang ditetapkan ($p > 0.05$).
<br> Tidak dapat disimpulkan bahwa proporsi nilai total Sulfat Dioxide yang lebih dari 40, adalah tidak sama dengan 50%.